### PR TITLE
fix LayoutInitializer to handle null pipeline and add MacroRunner particle+nodetype

### PR DESCRIPTION
**Pull request by mariakleiner**
and avoid outputting an unchanged pipeline in ConnectionUpdator.

### CODE REVIEW

File: pkg/Library/NodeGraph/ConnectionUpdator.js [hunk 1]
```diff
@@ -17,15 +17,17 @@ update(inputs, state) {
     const {pipeline, nodeTypes, candidates} = inputs;
     assign(state, {pipeline, candidates});
     if (candidates) {
-      //let changed = false;
```
Comment (sjmiles) [inline pkg/Library/NodeGraph/ConnectionUpdator.js:20]: maybe 

let changed = this.removePipeline...
changed ||= this.updatePiepline...

**Review by sjmiles — APPROVED** [review]

File: pkg/Library/NodeGraph/ConnectionUpdator.js (hunk 1 above)
Comment (mariakleiner) [reply pkg/Library/NodeGraph/ConnectionUpdator.js:20]: `change || = doSomething()` will not call `doSomething`, if `changed` is already `true`.

**Comment by mariakleiner** [comment]
sorry, i completely messed up LayoutInitializer again (didn't work for new pipelines), so I fixed it (hopefully) and also added the MacroRunner to the same PR.

File: pkg/Library/NodeGraph/ConnectionUpdator.js (hunk 1 above)
Comment (sjmiles) [reply pkg/Library/NodeGraph/ConnectionUpdator.js:20]: Touché

**Pull request merged**